src/tequila/circuit/gates.py

```diff
@@ -680,6 +680,146 @@ def CRz(control: int, target: int, angle: float) -> QCircuit:
     return Rz(target=target, control=control, angle=angle)
 
 
+def U(theta, phi, lambd, target: typing.Union[list, int], control: typing.Union[list, int] = None):
+    """
+    Notes
+    ----------
+    Convenient gate, one of the abstract gates defined by OpenQASM.
+
+    .. math::
+        U(\\theta, \\phi, \\lambda) = R_z(\\phi + 3\\pi)R_x(\\pi/2)R_z(\\theta + \\pi)R_x(\\pi/2)R_z(\\lambda)
```

Comment:
In case you know how the matrix looks, you could add that as well, might be helpful for some people

Reply:
I have added the matrix representation of the U gate